packages/flutter/lib/src/material/popup_menu.dart

```diff
@@ -839,7 +841,7 @@ class _PopupMenuRoute<T> extends PopupRoute<T> {
   }
 
   @override
-  Duration get transitionDuration => _kMenuDuration;
+  Duration get transitionDuration => animationDuration ?? _kMenuDuration;
```

Comment:
Hey, I think we can't rename it as `transitionDuration` since it's an overridden field and we can't declare a variable with the same name in the class.

Reply:
you can do this
```dart
class _PopupMenuRoute<T> extends PopupRoute<T> {
  _PopupMenuRoute({
    required this.position,
    required this.items,
    this.initialValue,
    this.elevation,
    this.surfaceTintColor,
    this.shadowColor,
    required this.barrierLabel,
    this.semanticLabel,
    this.shape,
    this.color,
    required this.capturedThemes,
    this.constraints,
    required this.clipBehavior,
    this.transitionDuration = _kMenuDuration,
    super.settings,
  }) : itemSizes = List<Size?>.filled(items.length, null),
       // Menus always cycle focus through their items irrespective of the
       // focus traversal edge behavior set in the Navigator.
       super(traversalEdgeBehavior: TraversalEdgeBehavior.closedLoop);

@override
 final Duration transitionDuration;
```